### ISSUE TITLE
CI is failing on master

### ISSUE THREAD

**Issue by mlubin** [JuliaOpt/JuMP.jl#2059]
https://travis-ci.org/JuliaOpt/JuMP.jl/jobs/582768312#L382

```
Automatic bridging with cache for bridged model: Test Failed at /home/travis/build/JuliaOpt/JuMP.jl/test/model.jl:197
  Expression: optimizer_index(x) === (MOI.get((((JuMP.backend(model)).optimizer).model).optimizer, MOI.ListOfVariableIndices()))[1]
   Evaluated: MathOptInterface.VariableIndex(12345676) === MathOptInterface.VariableIndex(12345679)
Stacktrace:
 [1] macro expansion at /home/travis/build/JuliaOpt/JuMP.jl/test/model.jl:197 [inlined]
 [2] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1083 [inlined]
 [3] macro expansion at /home/travis/build/JuliaOpt/JuMP.jl/test/model.jl:179 [inlined]
 [4] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1083 [inlined]
 [5] test_model() at /home/travis/build/JuliaOpt/JuMP.jl/test/model.jl:164
```

**Comment by odow** [comment]
The last merged PR ran CI on MOI 0.9.1, but by the time it was merged and CI run on master, MOI 0.9.2 had been released. This just looks like variables being added in a different order?

**Comment by odow** [comment]
Indeed, this is what seems to have happened. I would regard the failing test as flaky:
https://github.com/JuliaOpt/JuMP.jl/blob/3caea82df6d73744f9b060c04a81c77058f8011c/test/model.jl#L197
since it relies on internal details (like what order variables get added in some nested bridge/optimizer/layers) that JuMP shouldn't worry about.
